routing/endpointregistry.go

```diff
@@ -0,0 +1,103 @@
+package routing
+
+import (
+	"sync"
+	"time"
+)
+
+// metrics describe the data about endpoint that could be
+// used to perform better load balancing, fadeIn, etc.
+type metrics interface {
+	DetectedTime() time.Time
+	InflightRequests() int64
+}
+
+type entry struct {
+	detected         time.Time
+	inflightRequests int64
+}
+
+var _ metrics = &entry{}
+
+func (e *entry) DetectedTime() time.Time {
+	return e.detected
+}
+
+func (e *entry) InflightRequests() int64 {
+	return e.inflightRequests
+}
+
+type EndpointRegistry struct {
+	mu sync.Mutex
+
+	data map[string]*entry
+}
+
+var _ PostProcessor = &EndpointRegistry{}
+
+type RegistryOptions struct {
+}
+
+func (r *EndpointRegistry) Do(routes []*Route) []*Route {
+	for _, route := range routes {
+		for _, epi := range route.LBEndpoints {
+			metrics := r.GetMetrics(epi.Host)
+			if metrics.DetectedTime().IsZero() {
+				r.SetDetectedTime(epi.Host, time.Now())
+			}
+		}
+	}
+	return routes
+}
+
+func NewEndpointRegistry(o RegistryOptions) *EndpointRegistry {
+	return &EndpointRegistry{data: map[string]*entry{}}
+}
+
+func (r *EndpointRegistry) GetMetrics(key string) metrics {
```

Comment:
Did you hit this in your benchmark with a lot of entries (maybe 10k) in the registry?

Reply:
I added the code which adds a lot of entries in the benchmark, the results are essentially the same.